spraycharles.py

```diff
@@ -13,192 +13,517 @@
+            style="warning",
+        )
+        time.sleep(interval * 60)
+        print()
+        return 0, new_hit_total
 
+    else:
+        return login_attempts, total_hits
 
 
 def check_file_contents(file_path, current_list):
     new_list = []
     try:
-        with open(file_path, 'r') as f:
+        with open(file_path, "r") as f:
             new_list = f.read().splitlines()
     except:
+
         # file either no longer exists, or -p flag was given a password and not a file
         pass
-    
+
     additions = list(set(new_list) - set(current_list))
-    return additions    
+    return additions
 
 
 def print_attempt(username, password, response, csvfile):
-    if response == 'timeout':
-        code = 'TIMEOUT'
-        length = 'TIMEOUT'
+    if response == "timeout":
+        code = "TIMEOUT"
+        length = "TIMEOUT"
     else:
         code = response.status_code
         length = str(len(response.content))
-    print('%-27s %-17s %13s %15s' % (username, password, code, length))
-    output = open(csvfile, 'a')
-    output.write('%s,%s,%s,%s\n' % (username, password, code, length))
-    output.close() 
+    print("%-27s %-17s %13s %15s" % (username, password, code, length))
+    output = open(csvfile, "a")
+    output.write("%s,%s,%s,%s\n" % (username, password, code, length))
+    output.close()
 
 
-def login(target, username, password, csvfile):
+def login(target, username, password, csvfile, table):
+
+    # Getting class name real quick
+    name = type(target).__name__
+
     try:
         response = target.login(username, password)
-        target.print_response(response, csvfile)
+
+        # If module office365, we need to return more info
+        if name.lower() == "office365":
+            result, message, ruser, rpass, rcode, rlength = target.print_response(
+                response, csvfile
+            )
+
+        # If its anything else we just return the usual
+        else:
+            ruser, rpass, rcode, rlength = target.print_response(response, csvfile)
+
     except requests.ConnectTimeout as e:
-        target.print_response(response, csvfile, timeout=True)
+
+        if name.lower() == "office365":
+            result, message, ruser, rpass, rcode, rlength = target.print_response(
+                response, csvfile, timeout=True
+            )
+
+        else:
+            ruser, rpass, rcode, rlength = target.print_response(response, csvfile)
+
     except (requests.ConnectionError, requests.ReadTimeout) as e:
-        colors.color_print('\n[!] Connection error - sleeping for 5 seconds', colors.red)
+        console.print("\n[!] Connection error - sleeping for 5 seconds", style="danger")
         sleep(5)
-        login(target, username, password, csvfile)
+        login(target, username, password, csvfile, table)
+
+    if name.lower() == "office365":
+        return result, message, ruser, rpass, rcode, rlength
+    else:
```

Comment:
SMB needs handled here

```suggestion
    elif name.lower() == "smb":
        return ruser, rpass, rcode
     else:
```